servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -413,6 +419,13 @@ private void removeBlockNoLock(long userId, long blockId) throws IOException {
     mMetaManager.removeBlockMeta(blockMeta);
   }
 
+  // always update the metadata view before doing freeSpacewithNewView
+  private BlockMetadataManagerView getUpdatedView() throws IOException {
+    // TODO: update the view object instead of creating new one every time
+    return new BlockMetadataManagerView(mMetaManager, mPinnedInodes,
+        mLockManager.getLockedBlocks());
+  }
```

Comment:
Do you need to consider concurrency with `setPinnedInodes`? Since its possible to get an empty set after the clear?